src/coreclr/src/jit/compiler.h

```diff
@@ -8272,21 +8277,63 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
         return false;
     }
 
-    bool compSupports(CORINFO_InstructionSet isa) const
+#ifdef DEBUG
+    // Answer the question: Is a particular ISA supported?
+    // Use this api when asking the question so that future
+    // ISA questions can be asked correctly or when asserting
+    // support/nonsupport for an instruction set
+    bool compIsaSupportedDebugOnly(CORINFO_InstructionSet isa) const
     {
 #if defined(TARGET_XARCH) || defined(TARGET_ARM64)
         return (opts.compSupportsISA & (1ULL << isa)) != 0;
 #else
         return false;
 #endif
     }
+#endif // DEBUG
 
     void notifyInstructionSetUsage(CORINFO_InstructionSet isa, bool supported) const;
 
+    // Answer the question: Is a particular ISA supported?
+    // The result of this api call will exactly match the target machine
+    // on which the function is executed (except for CoreLib, where there are special rules)
+    bool compExactlyDependsOn(CORINFO_InstructionSet isa) const
+    {
+
+#if defined(TARGET_XARCH) || defined(TARGET_ARM64)
+        uint64_t isaBit       = (1ULL << isa);
+        bool     isaSupported = (opts.compSupportsISA & (1ULL << isa)) != 0;
+        if ((opts.compSupportsISAReported & isaBit) == 0)
+        {
+            notifyInstructionSetUsage(isa, isaSupported);
```

Comment:
Maybe add a jitdump here? Eg
```suggestion
            JITDUMP("Notifying runtime that codegen for this method is %susable if runtime HW supports %s\n",
                            isaSupported ? "" : "NOT ", InstructionSetToString(isa));
            notifyInstructionSetUsage(isa, isaSupported);
```

Reply:
notifyInstructionSetUsage does a jitdump internally. (All calls to the api are dumped)